### PR TITLE
Svlt sap div

### PR DESCRIPTION
**Pull request by DivyaKrishnan26**
Auth Mod for sapindale

### CODE REVIEW

File: src/components/Nav.svelte [hunk 1]
```diff
@@ -51,15 +73,28 @@
...
+  float: right;     
+  } 
+  
 </style>
 
 <nav>
   <ul>
     <li>
       <a class:selected={segment === undefined} href="/.">Home</a>
     </li>
```
Comment (SaschaAdler) [inline src/components/Nav.svelte]: Unlike in `d10n`, you should be able to use [Svelte #if directives](https://svelte.dev/docs#if) to determine whether the login button or the welcome message should show up here, as opposed to using `visibility: hidden`. I might be wrong.

File: src/components/Nav.svelte [hunk 2]
```diff
@@ -1,5 +1,27 @@
...
+import {state} from '../auth.js';
+import {onMount} from 'svelte';
+let value;
   export let segment;
+
+  onMount(async () => {
+  
+    state.subscribe(checkstate => {
+     
+      console.log(checkstate);
```
Comment (SaschaAdler) [inline src/components/Nav.svelte]:
```suggestion
      // get rid of console.log in production code
```

File: src/auth.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import {get,readable} from 'svelte/store';
+var value;
```
Comment (SaschaAdler) [inline src/auth.js]: You should provide an initial value for what state should return, even if in practice it gets set to something else instantaneously. I'm also a bit uncomfortable with using status codes here. Maybe just set this to the "LOADING", "FAILED", "SUCCEEDED" states we drew up?

File: src/auth.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+export const state =  readable(value,function auth(set){
+fetch("https://auth.canadiana.ca/1/auth",{
+credentials:"include",
+})
+.then(function(response, data){
+    if(response.status == 200){
+       set(value = response.status);
```
Comment (SaschaAdler) [inline src/auth.js]: I don't think you need to set `value` within the call to `set` here.

```suggestion
       set(response.status);
```

File: src/auth.js [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+    }
+    else if(response.status == 401){
+        
+        set(value = response.status);
```
Comment (SaschaAdler) [inline src/auth.js]:
```suggestion
        set(response.status);
```

File: src/auth.js (hunk 3 above)
Comment (SaschaAdler) [reply src/auth.js]: 204 should be treated equally to 200, and doesn't indicate "PENDING". "PENDING" should be the default value of `value` before the auth fetch causes it to change. (i.e., change `var value;` above to `var value = "PENDING";`

Comment (SaschaAdler) [reply src/auth.js]: `amsa` might also return `403` instead of `401`, and that should be treated as a failure, also.

**Comment by DivyaKrishnan26** [comment]
Have treated the data differently. not sure about the repercussion about this particular way.

File: src/auth.js [hunk 6]
```diff
@@ -1,18 +1,21 @@
 import { get, readable } from "svelte/store";
-var value;
+var value = "LOADING";
+var obj;
```
Comment (SaschaAdler) [inline src/auth.js]: You've got an `obj` here you don't use.

File: src/auth.js [hunk 7]
```diff
@@ -0,0 +1,21 @@
+import { get, readable } from "svelte/store";
+var value = "LOADING";
+var obj;
+export const state = readable(value, function auth(set) {
+  fetch("https://auth.canadiana.ca/1/auth", {
+    credentials: "include"
+  })
+    .then(function(response) {
+      if (response.status >= 200 && response.status <= 300) {
```
Comment (SaschaAdler) [inline src/auth.js]:
```suggestion
      if (response.status >= 200 && response.status < 300) {
```

File: src/auth.js [hunk 8]
```diff
@@ -0,0 +1,21 @@
...
+        value = "SUCCESS";
+        set(value);
+      } else if (response.status >= 400 && response.status <= 500) {
```
Comment (SaschaAdler) [inline src/auth.js]: Anything above 400 is a failure.

```suggestion
      } else if (response.status >= 400) {
```

File: src/components/Nav.svelte [hunk 9]
```diff
@@ -51,15 +63,32 @@
...
     </li>
-    <li class="right">
-      <a href=".">Figure out auth button here</a>
-    </li>
+    {#if value[1] === 'SUCCESS' && value[2] !== 'undefined'}
+      <p id="Welcome">Logged in as:{name}</p>
+    {:else}
+      <li class="right" id="login">
+        <a
+          href="https://auth.canadiana.ca/1/azuread/login?redirectUrl=http://admin.canadiana.ca/"
```
Comment (SaschaAdler) [inline src/components/Nav.svelte]: The redirect URL should figure out the host programmatically, if possible.

**Comment by DivyaKrishnan26** [comment]
Have changed the cosmetic changes, for the URL to resolve i have right now added the port to avoid the Ubuntu page. Need a bit of research, on what to use as the location.origin or the location.referrer errors out with svelte/sapper/webpack.  Will keep looking on to it along with the usage of stores to hide the other tools based on authetication.

File: src/components/Nav.svelte [hunk 10]
```diff
@@ -1,5 +1,20 @@
...
+  import { state } from "../auth.js";
+  import { onMount } from "svelte";
+  var value = [];
+  var name;
+  var redirectUrl;
   export let segment;
+
+  onMount(async () => {
+    var ref = window.location.origin;
+    redirectUrl = ref;
```
Comment (SaschaAdler) [inline src/components/Nav.svelte]: You don't need a `ref` var here.

```suggestion
    redirectUrl = window.location.origin;
```

File: src/components/Nav.svelte [hunk 11]
```diff
@@ -51,15 +66,32 @@
...
 <nav>
   <ul>
     <li>
       <a class:selected={segment === undefined} href="/.">Home</a>
     </li>
-    <li class="right">
-      <a href=".">Figure out auth button here</a>
-    </li>
+    {#if value[1] === 'SUCCESS' && value[2] !== 'undefined'}
+      <p id="Welcome">Logged in as:{name}</p>
```
Comment (SaschaAdler) [inline src/components/Nav.svelte]:
```suggestion
      <p id="Welcome">Logged in as: {name}</p>
```

**Pull request merged**